### PR TITLE
Ensure we set a content-type for healthz

### PR DESCRIPTION
**Pull request by dims**
Change-Id: I453b1433c69bf26c28da873dbdd1ac25006b8d60

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

`/healthz` spits out plain text, so make sure we tell http clients that the body is plain text

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
Comment (tallclair) [inline staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go]:
```suggestion
 		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
 		w.Header().Set("X-Content-Type-Options", "nosniff")
```

**Comment by dims** [comment]
Thanks @tallclair accepted your suggestion

**Review by logicalhan — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
Comment (logicalhan) [inline staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go]: Can we move this line (and the one below it) to between lines 159 and 160? That should ensure that the content type is set for early returns from our handler (i.e. healthz failures).

**Comment by liggitt** [comment]
/hold

looks like a spurious commit got pushed...

File: staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go]: no, http.Error sets these headers itself

**Comment by dims** [comment]
whoops fixed @liggitt 

/hold cancel

File: staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
Comment (logicalhan) [reply staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go]: Ah, interesting.. 

Would it be worth considering modifying mux directly so that we don't have to set the nosniff each time? Maybe something like 'HandleWithMimeType' which always sets nosniff?

Comment (logicalhan) [reply staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go]: Mostly, this seems to be fixing the same issue as #72535, it seems possible we would run into this again in the future unless we set these things automatically.

Comment (dims) [reply staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go]: :) y i filed this first and then @tallclair saw it and followed up with #72535 yes, we could have merged both into a single PR

Comment (logicalhan) [reply staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go]: What about fixing this at the mux level? Then we would not have to set for each individual path.

**Comment by dims** [comment]
@liggitt @sttts can one of you please lgtm if appropriate?

**Comment by liggitt** [comment]
> FYI, we can hopefully use this change here (#72589).

that is more useful for handlers that don't have branches that delegate to http.Error and would set content-type and nosniff as the first two lines of the handler function

> @liggitt @sttts can one of you please lgtm if appropriate?

/lgtm
/approve

**Pull request merged**
